cmd/metacache-set.go

```diff
@@ -142,32 +146,22 @@ func (o *listPathOptions) gatherResults(in <-chan metaCacheEntry) func() (metaCa
 			if !o.IncludeDirectories && entry.isDir() {
 				continue
 			}
-			if debugPrint {
```

Comment:
This is not a transparent change. While it looks clunky it is by design.

A statement in `if false` will be completely removed by the compiler, whereas `debugPrint("gather got: %s", entry.name)` will always have to evaluate parameters AND all parameters are assumed to escape, so values can no longer be on stack.

Since thousands of objects can stream through here it could affect performance. You are welcome to change the `debugPrint` in ` *bucketMetacache.findCache()`. However don't assume that constants are propagated through function calls.

Reply:
This was just original implementation I just borrowed everywhere.

Reply:
Yes, but it shouldn't be in critical paths. And these are critical.

The change is much slower, in #11063 the difference was a 2-3x speedup just by not evaluating the parameters and letting the values escape.

Reply:
Btw the change is entirely different now can you take a look @klauspost ??